### PR TITLE
Metadata templates

### PR DESCRIPTION
**Pull request by dhatcher**
A couple of small bug fixes to allow editing in the 4.x JSAPI.

### CODE REVIEW

File: lib/query.js [hunk 1]
```diff
@@ -81,7 +81,11 @@ function geoservicesPostQuery (data, queriedData, params) {
       throw error
     }
     oids = oids.map(i => {
-      return parseInt(i)
+      if (isNaN(i)) {
+        return i
```
Comment (rgwozdz) [inline lib/query.js:85]: Can you cover this in a test?

File: lib/templates.js [hunk 2]
```diff
@@ -56,6 +56,7 @@ function renderLayer (data = {}, { params = {}, query = {} } = {}) {
     json.uniqueIdField.name = metadata.idField
   }
   if (metadata.capabilities) { json.capabilities = metadata.capabilities }
+  if (metadata.templates) { json.templates = metadata.templates }
```
Comment (rgwozdz) [inline lib/templates.js:59]: Test coverage?

File: lib/query.js (hunk 1 above)
Comment (dhatcher) [reply lib/query.js:85]: Test coverage added.

File: lib/templates.js (hunk 2 above)
Comment (dhatcher) [reply lib/templates.js:59]: Added a new test for this.

**Review by rgwozdz — APPROVED** [review]
Would say this is a patch or new functionality?

**Comment by dhatcher** [comment]
Well they are changes I had to make so editing will work with 4.21 + of the JSAPI. I was thinking patch. What do you think?

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rgwozdz** [comment]
Ok, patch makes sense then.

**Pull request merged**